extensions/amp-pinterest/0.1/pin-widget.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../src/url';
+
+import {Util} from './util';
+
+// Popup options
+const POP = 'status=no,resizable=yes,scrollbars=yes,' +
+  'personalbar=no,directories=no,location=no,toolbar=no,' +
+  'menubar=no,width=900,height=500,left=0,top=0';
+
+/*
+ * Pinterest Pin Widget
+ * @attr data-url: the source url for the Pin
+ **/
+export class PinWidget {
+
+  constructor(rootElement) {
+    AMP.assert(rootElement.getAttribute('data-url'),
+      'The data-url attribute is required for Pin widgets');
+    this.element = rootElement;
+  }
+
+  /*
+   * Override the default href click handling to log and open popup
+   * @param {Event} evt: the HTML event object
+   **/
+  handleClick(evt) {
+    evt.preventDefault();
+    const el = evt.target;
+    const shouldPop = el.getAttribute('data-pin-pop') || false;
+    const href = el.getAttribute('data-pin-href');
+    const log = el.getAttribute('data-pin-log');
+    if (href) {
+      if (shouldPop) {
+        window.open(href, '_pinit', POP);
+      } else {
+        window.open(`${href}?amp=1&guid=${Util.guid}`, '_blank');
+      }
+    }
+    if (log) {
+      Util.log('&type=' + log);
+    }
+  }
+
+  fetchPin() {
+    const baseUrl = 'https://widgets.pinterest.com/v3/pidgets/pins/info/?';
+    const query = `pin_ids=${this.pinId}&sub=www&base_scheme=https`;
+    return Util.fetch(baseUrl + query)
+      .then(response => {
+        try {
+          return response.data[0];
+        } catch (e) { return null; }
+      });
+  }
+
+  renderPin(pin) {
+    // start setting our class name
+    let className = '-amp-pinterest-embed-pin';
+    let imgUrl = assertHttpsUrl(pin.images['237x'].url);
+
+    // large widgets may come later
+    if (this.width === 'medium' || this.width === 'large') {
+      className = className + '-medium';
+      imgUrl = imgUrl.replace(/237/, '345');
+      Util.log('&type=pidget&pin_count_medium=1');
```

Comment:
This is OK with AMP as long as the client id is not passed down.
amp-pixel does the same thing.